File: Stack/JAVA/src/implementing_stack/Stack.java
package implementing_stack;

public class Stack {
	private int top;
	private int size;
	private int stack[];

	Stack(int size) {
		this.size = size; // stack's wanted size
		this.stack = new int[this.size];
		top = -1; // initialized to -1 in order to check whether the stack is empty or not
	}

	public void push(int newTop) {
		if (isFull()) {
			System.out.println("Stack is full. Can't push new value, " + newTop);
			return;
		}

		stack[++top] = newTop; // increase top by one then add newTop to stack[top]
		System.out.println("The new top integer is " + stack[top]);
	}

	public int pop() {
		if (isEmpty()) {
			System.out.println("Stack is empty.");
			return -1;
		}

		System.out.println("Pop and return top integer: " + stack[top] + "\nNew top integer is: " + stack[top - 1]);
		return stack[top--]; // return the top integer then reduce top by one
	}

	public int peek() {
		if (isEmpty()) {
			System.out.println("Stack is empty.");
			return -1;
		}

		System.out.println("Top integer is: " + stack[top]);
		return stack[top];
	}

	public boolean isEmpty() {
		return top == -1;
	}

	public boolean isFull() {
		return top == (size - 1);
	}

	public void printStack() {
		System.out.println("Printing stack: ");
		if (isEmpty()) {
			System.out.println("Stack is empty.");
			return;
		}
		
		for (int i = 0; i <= top; i++)
			System.out.print(stack[i] + "	");
		System.out.println();
	}

	public static void main(String[] args) {
		Stack stack = new Stack(3);

		stack.printStack();
		
		stack.push(1);
		stack.push(2);
		stack.push(3);

		stack.pop();

		stack.printStack();

		stack.peek();

		stack.push(3);
		stack.push(4);
		
		stack.printStack();
	}
}

File: Stack/Python/Stack.py
def is_empty(stack_len):
    return len(stack_len) == 0


def is_full(stack_len, stack_size):
    return len(stack_len) >= stack_size


def push(stack_app, new_top, stack_size):
    if is_full(stack_app, stack_size):
        print("Stack is full.")
        return
    stack_app.append(new_top)
    print("Item pushed: " + new_top)


def pop(stack_pop):
    if is_empty(stack_pop):
        print("Stack is empty. ", end='')
        return
    else:
        return stack_pop.pop()


def peek(stack_peek):
    if is_empty(stack_peek):
        print("Stack is empty. ", end='')
        return
    else:
        return stack_peek[-1:]


def print_stack(stack_print):
    print("Print stack: ", end='')
    for i in stack_print:
        print(i + "    ", end='')
    print()


size = int(input("Enter stack size: "))
stack = []

print(is_empty(stack))
print("Item popped: " + str(pop(stack)))
push(stack, str(11), size)
push(stack, str(12), size)
print_stack(stack)
push(stack, str(13), size)
print(is_full(stack, size))

push(stack, str(14), size)
print(is_full(stack, size))

print("Item popped: " + str(pop(stack)))
print("Item popped: " + str(pop(stack)))
push(stack, str(15), size)
print_stack(stack)
push(stack, str(16), size)
print_stack(stack)
print("Peek top item: " + str(peek(stack)))
